src/include/planner/plannodes/abstract_plan_node.h

```diff
@@ -42,29 +42,29 @@ class AbstractPlanNode {
      * @param output_schema output schema for plan node
      * @return builder object
      */
-    ConcreteType &SetOutputSchema(const std::shared_ptr<OutputSchema> &output_schema) {
-      output_schema_ = output_schema;
+    ConcreteType &SetOutputSchema(std::unique_ptr<OutputSchema> output_schema) {
+      output_schema_ = std::move(output_schema);
       return *dynamic_cast<ConcreteType *>(this);
     }
 
    protected:
     /**
      * child plans
      */
-    std::vector<std::shared_ptr<AbstractPlanNode>> children_;
+    std::vector<std::unique_ptr<AbstractPlanNode>> children_;
     /**
      * schema describing output of the node
      */
-    std::shared_ptr<OutputSchema> output_schema_;
+    std::unique_ptr<OutputSchema> output_schema_;
   };
 
   /**
    * Constructor for the base AbstractPlanNode. Derived plan nodes should call this constructor to set output_schema
    * @param children child plan nodes
    * @param output_schema Schema representing the structure of the output of this plan node
    */
-  AbstractPlanNode(std::vector<std::shared_ptr<AbstractPlanNode>> &&children,
-                   std::shared_ptr<OutputSchema> output_schema)
+  AbstractPlanNode(std::vector<std::unique_ptr<AbstractPlanNode>> &&children,
```

Comment:
If you're using the move constructor for children__ shouldn't the pattern be `std::unique_ptr<std::vector<.... `?